internal/pkg/template/templates/environment/cf.yml

```
@@ -71,6 +71,15 @@ Resources:
       Configuration:
         ExecuteCommandConfiguration:
           Logging: DEFAULT
+{{- if .Telemetry }}
+      ClusterSettings:
+        - Name: containerInsights
```

Will this accidentally disable an enabled container insights during `env upgrade`? What about 
```suggestion
    {{ - if .Telemetry.EnableContainerInsights}}
        - Name: containerInsights
           Value: enabled
     {{- end}}
```
?

---

1. For existing client environments: `.Telemetry` is nil so we have to have a whole `{{- if .Telemetry}}` block in order to not panic while upgrading their environments.
2. For new environments, `.Telemetry` is always not nil and `EnableContainerInsights` is either `true` or `false` so it should be fine.

Does that make sense or am I missing a case?

---

My concern was that it will disable the container insight if the block (i.e. the block of `ClusterSettings`) isn't rendered into the template. But this probably isn't a valid concern because for existing environments the template has always looked like this (without the block).